static/src/components/setting/index.vue

```diff
@@ -0,0 +1,387 @@
+<template>
+  <div class="stats-config row">
+    <div class="col-md-12">
+      <div
+        v-for="category in inputFields"
+        :key="category.display"
+        class="form-group row"
+      >
+        <div class="col-md-5">
+          <p> {{ category.display }} </p>
+        </div>
+        <div class="col-md-7 pull-right">
+          <div
+            v-for="field in category.fields"
+            :key="field.name"
+          >
+            <input
+              v-if="field.type==='TextField'"
+              v-model="externalConfigDict[field.name]"
+              type="text"
+              class="form-control input-sm"
+            >
+            <select
+              v-if="field.type==='SelectField'"
+              v-model="externalConfigDict[field.name]"
+              class="form-control input-sm"
+            >
+              <option
+                v-for="choice in field.choices"
+                :key="choice[0]"
+                :value="choice[0]"
+              >
+                {{ choice[1] }}
+              </option>
+            </select>
+          </div>
+        </div>
+      </div>
+      <div
+        v-if="privateInstance"
+        class="form-group row"
+      >
+        <div class="col-md-5">
+          <p> data access </p>
+        </div>
+        <div class="col-md-7 pull-right">
+          <div class="form-check">
+            <span
+              v-for="(lvalue, level) in accessLevels"
+              :key="level"
+              :value="lvalue"
+            >
+              <input
+                v-model="accessLevels[level]"
+                class="form-check-input"
+                type="checkbox"
+              >
+              <label
+                class="form-check-label"
+                for="defaultCheck1"
+              >
+                <p> {{ level }} &nbsp;</p>
+              </label>
+            </span>
+          </div>
+        </div>
+      </div>
+      <div
+        v-if="privateInstance"
+        class="form-group row"
+      >
+        <div class="col-md-5">
+          <p> assign users </p>
+        </div>
+        <div class="col-md-7 pull-right">
+          <table
+            class="table table-striped table-hover"
+            style="border: 1px ridge silver"
+          >
+            <thead>
+              <tr style="border-top: 1px ridge silver">
```

Comment:
Can we avoid inline style? Can part of this stile be made generic and reused for consistency across pages?